### PR TITLE
feat(custom-scm): Add base for Custom SCM integration 

### PR DESCRIPTION
**Pull request by MeredithAnya**
**History of Repositories @ Sentry:**
Prior to the introduction of first party integrations (global on the organization), we had our plugins. At first, the source code plugins like GitHub only had issue tracking. When we first added `Repositories` to Sentry, it was done as a separate step to installing a project level plugin because repositories were on the organization. 

This is why we currently have **Settings > Repositories** page, you used to add repositories via that page.

Once we moved GitHub (and other repo based plugins) to top level integrations, adding repositories happened under the configuration **Settings > Integrations > GitHub > Configure**, and you could no longer add repositories the old way (you had to upgrade to using the integration not the plugin).

Why do we still have that page?
1. Only place that shows repositories tied to the Legacy plugins
1. Only place that shows repositories that were created by **manual commit tracking** (These are the repos with "Unknown Provider")

Repositories are the basis for commit tracking, and are required. We have allowed people to use the API to send us their commits with releases, and now make it easier by allowing you to [send commits via the sentry-cli](https://docs.sentry.io/product/cli/releases/#alternatively-without-a-repository-integration).

**Custom Source Control Management (SCM) Integration**
As we build more features on top of repositories (*Stack Trace Linking*, *Codeowners*), we want to make sure those who can't add a first party integration can still use some those features. By creating a sort of "dummy" integration we can:
* Reuse almost all the UI components for other repo-based integrations
* Create/Edit their Code Mappings, User Mappings, Team Mappings in the UI instead of only via the API
* Give repositories an explicit provider instead of being 'Unknown'

**The Flow <sup>TM</sup>**
1. The Custom SCM will be listed in the integrations directory just like everything else
1. The **Add Installation** button will open up a django view similarly to how GitLab and GHE work. See https://github.com/getsentry/sentry/pull/25789#issuecomment-830466392.
1. Once the form is submitted, the integration is created using the `name` and `url` provided, an `external_id` is automatically generated for them. See https://github.com/getsentry/sentry/pull/25789#issuecomment-830466544. The `url` and the `name` will be used to form the domain name, they will be able to update both these values in this PR https://github.com/getsentry/sentry/pull/25803/.
1. Adding a repository doesn't have the full functionality yet, but it will be the process of adding 'unknown provider' repos to the custom scm integration. In this PR, I've just implemented the `get_repositories`. See https://github.com/getsentry/sentry/pull/25789#issuecomment-830466696. Future PR here https://github.com/getsentry/sentry/pull/25801 to add the ability to edit the `name` and `url` for repositories tied to the custom scm to allow stack trace linking.

Todos
- [x] Feature flag this `organizations:integrations-custom-scm`
- [x] Testing
- [X] Figure out if/how giving the repo a provider affects uploading commits, do I need to add the provider to the `sentry-cli`
    * all signs point to no, I shouldn't need to, but will test this out once I have the rest of the integration PRs merged

### CODE REVIEW

File: src/sentry/integrations/manual_source_control/integration.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    def get_repositories(self, query=None):
+        """
+        Used to get any repositories that are not already tied
+        to an integration.
+        """
+        repos = Repository.objects.filter(
+            organization_id=self.organization_id,
+            provider__isnull=True,
+            integration_id__isnull=True,
+            status=0,
```
Comment (MeredithAnya) [inline src/sentry/integrations/manual_source_control/integration.py:63]: should use `ObjectStatus`

**Comment by MeredithAnya** [comment]
**Step 2.**

> Screen Shot 2021-04-30 at 4 49 43 PM

**Comment by MeredithAnya** [comment]
**Step 3.**

> Screen Shot 2021-04-30 at 4 50 08 PM

**Comment by MeredithAnya** [comment]
**Step 4:**

>Screen Shot 2021-04-30 at 4 50 28 PM

File: src/sentry/integrations/custom_scm/integration.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+FEATURES = [
+    FeatureDescription(
+        """
+        Send your own commits
+        """,
+        IntegrationFeatures.COMMITS,
+    ),
+    FeatureDescription(
+        """
+        Stack trace linky dink
```
Comment (NisanthanNanthakumar) [inline src/sentry/integrations/custom_scm/integration.py:34]: @MeredithAnya we probs should have some real copy here

Comment (MeredithAnya) [reply src/sentry/integrations/custom_scm/integration.py:34]: yeah I'm going just wait until the end for this, I don't think we need to do this now

File: tests/sentry/api/endpoints/test_organization_config_integrations.py [hunk 3]
```diff
@@ -27,13 +27,13 @@ def test_provider_key(self):
         assert response.data["providers"][0]["name"] == "Example Server"
 
     def test_feature_flag_integration(self):
-        features.add("organizations:integrations-feature_flag_integration", OrganizationFeature)
+        features.add("organizations:integrations-feature-flag-integration", OrganizationFeature)
```
Comment (NisanthanNanthakumar) [inline tests/sentry/api/endpoints/test_organization_config_integrations.py:30]: @MeredithAnya I think we need to update https://github.com/getsentry/sentry/blob/7df6bbf4dffb707e5bfccb0cffca4b42b058d97e/tests/acceptance/test_organization_integration_detail_view.py#L20
for consistency

**Comment by NisanthanNanthakumar** [comment]
@MeredithAnya do we need `tests/sentry/integrations/custom_scm/__init__.py`? Its an empty file.

**Review by NisanthanNanthakumar — APPROVED** [review]

**Comment by MeredithAnya** [comment]
> @MeredithAnya do we need `tests/sentry/integrations/custom_scm/__init__.py`? Its an empty file.

I think it's quite common to have an empty `__init__.py` file, and is just partly how Python importing works (though it might not be needed in Python 3.3 it looks like?). I found https://docs.python.org/3/reference/import.html#packages helpful

**Pull request merged**